ifrs17/Utils/ActivityLogsAndImports.ipynb

```diff
@@ -0,0 +1,302 @@
+                "\n    }",
+                "\n",
+                "\n    private string[] GetInvolvedPartitions(ImportOptions options)",
+                "\n    {",
+                "\n        //Andrey Katz: Get all the relevant partitions here ",
+                "\n        return null;",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record KeyedStringImport : ImportWithKey",
+                "\n{",
+                "\n    public string Content {get; init;}",
+                "\n",
+                "\n    public KeyedStringImport(StringImportOptions options)",
+                "\n    {",
+                "\n        importOptions = options;",
+                "\n        Id = Guid.NewGuid();",
+                "\n        Content = options.Content;",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record KeyedDataSetImport : ImportWithKey",
+                "\n{",
+                "\n    public KeyedDataSetImport(DataSetImportOptions options)",
+                "\n    {",
+                "\n        importOptions = options;",
+                "\n        Id = Guid.NewGuid();",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record KeyedStreamImport : ImportWithKey",
+                "\n{",
+                "\n    public KeyedStreamImport(StreamImportOptions options)",
+                "\n    {",
+                "\n        importOptions = options;",
+                "\n        Id = Guid.NewGuid();",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<ActivityLog> ExecuteWithStoreActivityAsync(this ImportOptionsBuilder builder, ",
+                "\n                                                                                ISessionVariable session, ",
+                "\n                                                                                IDataSource dataSource)",
+                "\n{",
+                "\n    var log = await builder.ExecuteAsync();",
+                "\n    var options = builder.GetImportOptions();",
+                "\n    var activity = new KeyedActivity(log, session);",
+                "\n    switch(options)",
```

Comment:
I prefer the most recent syntax for the switch cases. 
See https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/patterns#declaration-and-type-patterns

Reply:
To the best of my knowledge this format is for one-liners, so I would leave this as it is.